examples/pytorch/language-modeling/run_clm.py

```diff
@@ -279,27 +279,30 @@ def main():
     # In distributed training, the load_dataset function guarantee that only one local process can concurrently
     # download the dataset.
     if data_args.dataset_name is not None:
+        kwargs = dict(
+            cache_dir=model_args.cache_dir,
+            use_auth_token=True if model_args.use_auth_token else None,
+        )
+        if data_args.streaming:
+            kwargs.update(streaming=True)
         # Downloading and loading a dataset from the hub.
         raw_datasets = load_dataset(
             data_args.dataset_name,
             data_args.dataset_config_name,
-            cache_dir=model_args.cache_dir,
-            use_auth_token=True if model_args.use_auth_token else None,
+            **kwargs,
```

Comment:
Just add `streaming = data_args.streaming` here.

Reply:
`streaming`-mode (`IterableDataset`) doesn't accept many of the non-streaming args. So your suggestion won't work.

Do you want me to make 2 `load_dataset` in a conditional instead?

Reply:
```
        if data_args.streaming:
            raw_datasets = load_dataset(
                data_args.dataset_name,
                data_args.dataset_config_name,
                streaming=True
            )
        else:
            raw_datasets = load_dataset(
                data_args.dataset_name,
                data_args.dataset_config_name,
                cache_dir=model_args.cache_dir,
                use_auth_token=True if model_args.use_auth_token else None,
            )
```

Reply:
ah, wait, I remembered wrong I think - it's the `map` that has different args. https://github.com/huggingface/transformers/pull/21343/files#r1091016058